anvil.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+set -e
+
+PORT=8545
+HOST="0.0.0.0"
+SLEEP_DURATION=0.1
+
+# Start the Anvil server.
+anvil --host "$HOST" --rpc-url "$MAINNET_RPC_URL" &
+ANVIL_PID=$!
+
+# Wait until Anvil is ready
+while ! nc -z localhost "$PORT"; do sleep "$SLEEP_DURATION"; done
+
+# Enable auto-impersonation
+cast rpc anvil_autoImpersonateAccount true
+
+# Deploy contracts, forking from the provided RPC URL
+forge script script/Deploy.s.sol --rpc-url "http://localhost:$PORT" --unlocked --broadcast --sender "$DEPLOYER"
+
+# Disable auto-impersonation
+cast rpc anvil_autoImpersonateAccount false
+
+# Wait for Anvil
+wait $ANVIL_PID
```

Comment:
`wait` is tricky. It will generally work but since `bash` is handling the initial signal and forwarding the signal to the background processes, we might not always get the behavior we want.

This may be totally fine in this context, but sometimes it can lead to confusing behavior.

A way around this is to re-exec to replace the process entirely with the desired command, so:

```bash
CMD="anvil --host $HOST --rpc-url $MAINNET_RPC_URL" --state /var/lib/anvil/state
...
$CMD & # Start anvil in the background
ANVIL_PID=$!

... (deploy the contracts) ...

kill $ANVIL_PID # Stop anvil
exec $CMD # Start anvil again, but replacing this process so it receives all signals directly instead of bash
```

Alternatively, you could create a separate container via a `docker-compose.yml` file which performs the contract deployment, keeping your base Anvil container simple (i.e. it just starts `anvil` and does nothing else).